konfera/models/ticket.py

```diff
@@ -44,7 +45,14 @@ def discount_calculator(self):
             return self.type.price * self.discount_code.discount / 100
         return 0
 
+    def clean(self):
+        if self.discount_code and self.discount_code.ticket_type != self.type:
+            raise ValidationError(
+                {'discount_code': _('Discount code is not applicable for the selected ticket type')})
+        super(Ticket, self).clean()
```

Comment:
Hmm I didn't realize that you didn't need to pass the class + context in python3! TIL. I'll make the change.